src/main/java/edu/tamu/cap/model/Schema.java

```diff
@@ -46,6 +53,15 @@ public Schema(String name, String namespace, String abbreviation) {
         setProperties(new ArrayList<Property>());
     }
 
+    private void addNamespace(String namespace) {
+        namespaces.add(namespace);
+    }
+
+    private String getNamespaceFromProperty(Property property) {
+        String uri = property.getUri();
+        return uri.split("#")[0];
```

Comment:
This will not work for all schemas. Some schemas have field name as fragment and some use path parameter.

Regex would be the most elegant. But I have solved this like:

```uri.substring(0, uri.lastIndexOf(uri.contains(HASH_TAG) ? HASH_TAG : FORWARD_SLASH))```